src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -77,9 +87,54 @@ public JsonParser(TypeToken<T> type) {
         this.type = type;
         this.gson = new GsonBuilder()
                 .registerTypeAdapter(Date.class, new DateAdapter().nullSafe())
+                .registerTypeAdapter(StartTime.class,
+                        new JobStartTimeJsonAdapter().nullSafe())
                 .registerTypeAdapter(Stats.class, new StatsDeserializer())
                 .registerTypeAdapter(Arguments.class, new ArgumentsDeserializer())
                 .create();
+        this.startJobDateFormat = JsonParser.START_TIME_FORMAT;
+    }
+
+    /**
+     * Creates a new JsonParser for the given type. Server's time zone is
+     * specified in the @{link ClientConfig}.
+     *
+     * @param type A TypeToken describing the type this parser produces.
+     * @param config ClientConfig used to fetch master's time zone
+     */
+    private JsonParser(TypeToken<T> type, ClientConfig config) {
+        this.type = type;
+        this.gson = new GsonBuilder()
+                .registerTypeAdapter(Date.class, new DateAdapter().nullSafe())
+                .registerTypeAdapter(StartTime.class,
+                        new JobStartTimeJsonAdapter().nullSafe())
+                .registerTypeAdapter(Stats.class, new StatsDeserializer())
+                .registerTypeAdapter(Arguments.class, new ArgumentsDeserializer())
+                .create();
+
+        synchronized (START_TIME_FORMAT) {
+            this.startJobDateFormat = (SimpleDateFormat) START_TIME_FORMAT.clone();
```

Comment:
OK. I've changed this now. This entire copy constructor is actually useless and it arose from me not initially understanding the flow of the program (how SaltStackClient uses JsonParser)

The simplest approach is probably the best - 1 instance of format per JsonParser instance.